src/Cli/dotnet/commands/dotnet-workload/list/WorkloadListCommand.cs

```diff
@@ -1,59 +1,74 @@
 // Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.Collections.Generic;
 using System.CommandLine.Parsing;
 using System.IO;
-using System.Linq;
 using System.Text.Json;
+using Microsoft.Deployment.DotNet.Releases;
 using Microsoft.DotNet.Cli;
 using Microsoft.DotNet.Cli.Utils;
-using Microsoft.DotNet.Configurer;
+using Product = Microsoft.DotNet.Cli.Utils.Product;
 using Microsoft.DotNet.Workloads.Workload.Install;
+using Microsoft.DotNet.Workloads.Workload.Install.InstallRecord;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using System.Linq;
 
 namespace Microsoft.DotNet.Workloads.Workload.List
 {
     internal class WorkloadListCommand : CommandBase
     {
+        private readonly IReporter _reporter;
+        private readonly VerbosityOptions _verbosity;
         private readonly bool _machineReadableOption;
-
-        private readonly string _mockInstallDirectory = Path.Combine(CliFolderPathCalculator.DotnetUserProfileFolderPath,
-            "DEV_mockworkloads");
+        private readonly IWorkloadInstallationRecordRepository _workloadRecordRepo;
+        private readonly SdkFeatureBand _sdkFeatureBand;
 
         public WorkloadListCommand(
-            ParseResult result
-        )
-            : base(result)
+            ParseResult result,
+            IReporter reporter = null,
+            IWorkloadInstallationRecordRepository workloadRecordRepo = null,
+            string version = null) : base(result)
         {
+            _reporter = reporter ?? Reporter.Output;
             _machineReadableOption = result.ValueForOption<bool>(WorkloadListCommandParser.MachineReadableOption);
+            _verbosity = result.ValueForOption<VerbosityOptions>(WorkloadListCommandParser.VerbosityOption);
+
+            var sdkVersion = new ReleaseVersion(version ?? Product.Version);
+            var dotnetPath = Path.GetDirectoryName(Environment.ProcessPath);
```

Comment:
nit: later we might want to have an abstraction to control all install paths. So far we have the same calculation for each command. dotnet tools have a similar problem. We later created IToolPackageStore.